_aggregations/pipeline/serial-diff.md

```diff
@@ -0,0 +1,292 @@
+         },
+         "aggs": {
+            "total_bytes": {
+               "sum": {
+                  "field": "bytes"     
+               }
+            },
+            "monthly_bytes_change": {
+               "serial_diff": {                
+                  "buckets_path": "total_bytes",
+                  "lag": 1
+               }
+            }
+         }
+      }
+   }
+}
+```
+{% include copy-curl.html %}
+
+The response contains the month-to-month difference for the second and third month. (The first month `serial_diff` cannot be calculated because there's no previous month to compare with):
+
+```json
+{
+  "took": 3,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 10000,
+      "relation": "gte"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "monthly_bytes": {
+      "buckets": [
+        {
+          "key_as_string": "2025-03-01T00:00:00.000Z",
+          "key": 1740787200000,
+          "doc_count": 480,
+          "total_bytes": {
+            "value": 2804103
+          }
+        },
+        {
+          "key_as_string": "2025-04-01T00:00:00.000Z",
+          "key": 1743465600000,
+          "doc_count": 6849,
+          "total_bytes": {
+            "value": 39103067
+          },
+          "monthly_bytes_change": {
+            "value": 36298964
+          }
+        },
+        {
+          "key_as_string": "2025-05-01T00:00:00.000Z",
+          "key": 1746057600000,
+          "doc_count": 6745,
+          "total_bytes": {
+            "value": 37818519
+          },
+          "monthly_bytes_change": {
+            "value": -1284548
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+The following line chart shows the results of the `serial_diff` aggregation. The x-axis represents time, and the y-axis shows the month-over-month change in total bytes transferred. Each data point on the line reflects the difference between the total bytes in that month and the previous month. For example, a value of 5,000,000 means the system transferred 5 million more bytes than the prior month; a negative value indicates a decrease. The first month is excluded from the line because there’s no earlier bucket to compare against (the difference is undefined). The line starts with the second month and continues across all available data. 
+
+![Example serial difference aggregation visualization]({{site.url}}{{site.baseurl}}/images/serial-diff-agg-result.png)
+
+This visualization helps you quickly spot spikes, drops, or trends in data volume over time.
+
+## Example: Multi-period differences
+
+Use a larger `lag` value to compare with buckets further back in time. The following example computes differences on weekly byte data with a lag of 4 (comparing each bucket with the bucket 4 weeks back). This has the effect of removing any variation with a period of four weeks:
+
```

Comment:
Line 130: "compare with buckets further back in time" needs a little revision for clarity, as does "the bucket 4 weeks back".

Reply:
Reworded.